tests/unit/ssg-module/test_oval_object_model/test_load_and_store.py

```diff
@@ -0,0 +1,399 @@
+    return oval_document.objects.get("oval:ssg-obj_sshd_rekey_limit:obj:1")
+
+
+@pytest.fixture
+def rpm_info_object(oval_document):
+    return oval_document.objects.get(
+        "oval:ssg-obj_test_package_openssh-server_installed:obj:1"
+    )
+
+
+@pytest.fixture
+def local_variable(oval_document):
+    return oval_document.variables.get("oval:ssg-sshd_line_regex:var:1")
+
+
+@pytest.fixture
+def external_variable(oval_document):
+    return oval_document.variables.get("oval:ssg-var_rekey_limit_size:var:1")
+
+
+@pytest.fixture
+def text_file_content_state(oval_document):
+    return oval_document.states.get("oval:ssg-state_sshd_rekey_limit:ste:1")
+
+
+@pytest.fixture
+def variable_state(oval_document):
+    return oval_document.states.get("oval:ssg-state_sshd_not_required:ste:1")
+
+
+# Tests:
+
+
+def test_load_oval_document(oval_document):
+    assert "oval:ssg-sshd_rekey_limit:def:1" in oval_document.definitions
+    assert "oval:ssg-test_sshd_rekey_limit:tst:1" in oval_document.tests
+    assert "oval:ssg-obj_sshd_rekey_limit:obj:1" in oval_document.objects
+    assert "oval:ssg-var_rekey_limit_time:var:1" in oval_document.variables
+    assert "oval:ssg-state_sshd_required:ste:1" in oval_document.states
+
+
+def _assert_oval_document_str(oval_document_str):
+    assert "oval:ssg-sshd_rekey_limit:def:1" in oval_document_str
+    assert "oval:ssg-test_sshd_rekey_limit:tst:1" in oval_document_str
+    assert (
+        "oval:ssg-obj_test_package_openssh-server_installed:obj:1" in oval_document_str
+    )
+    assert "combine_ovals.py from SCAP Security Guide" in oval_document_str
+    assert 'var_ref="oval:ssg-var_rekey_limit_size:var:1" ' in oval_document_str
+    assert "<ind:filepath>/etc/ssh/sshd_config</ind:filepath>" in oval_document_str
+
+
+def test_get_xml_element(oval_document):
+    oval_document_el = oval_document.get_xml_element()
+    oval_document_str = ElementTree.tostring(oval_document_el).decode()
+    _assert_oval_document_str(oval_document_str)
+
+
+def test_save_as_xml(oval_document):
+    oval_doc_path = os.path.join(TEST_BUILD_DIR, "oval.xml")
+    with open(oval_doc_path, "wb") as fd:
+        oval_document.save_as_xml(fd)
+
+    with open(oval_doc_path, "r") as fd:
+        oval_document_str = fd.read()
+        _assert_oval_document_str(oval_document_str)
+
+
+def test_load_shorthands(oval_document_from_shorthand):
+    assert "service_chronyd_or_ntpd_enabled" in oval_document_from_shorthand.definitions
+    assert "sudo_remove_nopasswd" in oval_document_from_shorthand.definitions
+
+    assert "test_nopasswd_etc_sudoers" in oval_document_from_shorthand.tests
+    assert "test_nopasswd_etc_sudoers_d" in oval_document_from_shorthand.tests
+
+    assert "object_nopasswd_etc_sudoers_d" in oval_document_from_shorthand.objects
+    assert "object_nopasswd_etc_sudoers" in oval_document_from_shorthand.objects
+
+    assert not oval_document_from_shorthand.variables
+    assert not oval_document_from_shorthand.states
```

Comment:
Wouldn't it be better to have some variables and states in the document as well? That would help you test them as well.

Reply:
Yes, I will prepare dog-food and extend the test.